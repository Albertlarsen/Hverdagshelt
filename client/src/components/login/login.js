// @flow

import ReactDOM from 'react-dom';
import * as React from 'react';
import { Component } from 'react-simplified';
import {User} from "../../classTypes";

export class Login extends Component{

}//end class
