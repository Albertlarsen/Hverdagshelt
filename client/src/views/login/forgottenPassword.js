//@flow
import { MailService } from "../../services";
import { Component } from "react";
import * as React from "react";
import ReactDOM from "react-dom";
import { Button, Form, FormControl, FormGroup } from "react-bootstrap";
import { history } from "../../index";

let mailService = new MailService();

export class ForgottenPassword extends Component<Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      email: { to: "" }
    };
  }

  handleStringChange = () => (event: SyntheticInputEvent<HTMLInputElement>) => {
    this.setState({
      email: { to: event.target.value }
    });
  };

  render() {
    return (
      <div className="container">
        <h2>Glemt passord</h2>
        <p>Skriv inn eposten din og få tilsendt et nytt passord på mail.</p>
        <Form>
          <FormGroup validationState={this.getValidationStateEmail()}>
            <div className="textfield">
              <FormControl
                value={this.state.email.to}
                onChange={this.handleStringChange()}
                placeholder="Epost"
              />
              <FormControl.Feedback />
            </div>
          </FormGroup>
          <Button bsStyle="primary" onClick={this.resetPassword}>
            Tilbakestill passord
          </Button>
        </Form>
      </div>
    );
  }

  getValidationStateEmail() {
    var validator = require("email-validator");
    const length = this.state.email.to.length;
    const bool = validator.validate(this.state.email.to);
    if (length == 0) return;
    else if (!bool) return "warning";
    else if (bool) return "success";
  }

  resetPassword = () => {
    mailService.sendResetPasswordMail({ to: this.state.email.to });
    history.push("/login");
  };
}
