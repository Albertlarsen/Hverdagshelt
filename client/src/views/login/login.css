.loginBox {

 }


.picture {
    max-height: 25em;
}