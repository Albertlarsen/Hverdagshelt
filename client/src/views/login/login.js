// @flow

import ReactDOM from "react-dom";
import * as React from "react";
import { Component } from "react";
import { User } from "../../classTypes";
import { UserService } from "../../services";
import { Alert, FormGroup, Form, FormControl, Button, Grid, Image, Row, Col } from "react-bootstrap";
let jwt = require("jsonwebtoken");
import login from "./login.css";
import { history } from "../../index";

let userService = new UserService();
const bcrypt = require("bcrypt-nodejs");

interface State {
  error: boolean;
  email: string;
  password: string;
  storedPassword: string;
  countyId: 0;
  string: "./logo.png";
  openPassword: string;
  blocked: boolean;
} //end interface

interface Props {
  notLoggedIn: boolean;
} //end interface

export class Login extends Component<Props, State> {
  state = {
    error: false,
    blocked: false,
    email: "",
    password: "",
    storedPassword: "",
    countyId: 0,
    string: "./logo.png",
    openPassword: "password"
  };

  handleChangeEmail(event: SyntheticEvent<HTMLButtonElement>) {
    this.setState({
      email: event.target.value
    });
  };

  handleChangePassword(event: SyntheticEvent<HTMLButtonElement>) {
    this.setState({
      password: event.target.value
    });
  };

  handleClickPassword=()=>{
    if(this.state.openPassword == "text"){
      this.setState({openPassword: "password"})
    }else{
      this.setState({openPassword: "text"})
    }
  };



  render() {
      window.onload = function(){
      if(!window.location.hash){
        window.location = window.location + '#loaded';
        window.location.reload();
      }//end condition
    };

    let alert_login;

    if (this.state.error) {
      alert_login = (
        <Alert bsStyle="danger">
          <h6>Brukernavn eller passord er feil. Prøv igjen!</h6>
        </Alert>
      );
    } else if(this.state.blocked){
      alert_login = (
        <Alert bsStyle={"danger"}>
          <h5> Du er blokkert fra å bruke hverdagshelt!
            Dersom du mener det har skjedd en feil, kontakt oss på hverdagshelt.scrum@gmail.com
          </h5>
        </Alert>
      );
    }
    else {
      alert_login = <p />;
    }

    let alert_notLoggedIn;
    if (this.props.notLoggedIn) {
      alert_notLoggedIn = confirm("Du må være logget inn for å gå videre");
    } else {
      <p />;
    } //end condition


    let changeIcon;

    if (this.state.openPassword == "text") {
      changeIcon = (<i className="fas fa-eye"></i>);
    } else {
      changeIcon = (<i className="fas fa-eye-slash"></i>);
    }


    return (
      <div className="login">
        <Grid>
          <Form>
            <Col sm={2} md={3} />

            <Col sm={8} md={6}>
              <div align="center">
                <Image
                  className="picture"
                  src={"./resources/logo_white.png"}
                  rounded
                />
              </div>

              <div className="loginBox">
                <Row className="show-grid">
                  <FormGroup>
                    <FormControl
                      type="text"
                      id="mailText"
                      placeholder="Email"
                      value={this.state.email}
                      onChange={this.handleChangeEmail.bind(this)}
                    />
                  </FormGroup>
                </Row>

                <Row className="show-grid" align="center">
                  <FormGroup>
                    <FormControl
                      type={this.state.openPassword}
                      id="passText"
                      placeholder="Passord"
                      value={this.state.password}
                      onChange={this.handleChangePassword.bind(this)}
                    />
                  </FormGroup>

                </Row>

                <Row className="show-grid" align="center">
                  <Button id="saveButton" type="button" onClick={this.save} bsStyle="primary">
                    Login
                  </Button>
                  <Button type="button" onClick={()=> this.handleClickPassword()}>{changeIcon}</Button>
                  {alert_login}
                  {alert_notLoggedIn}
                </Row>

                <div align="center">
                  <p>Har du ikke bruker?</p>

                  <p>Registrer deg <a href={'/#/registrer/privat'}>her hvis du er privatperson</a>, og <a
                    href="/#/registrer/bedrift"> her hvis du er

                    bedrift.</a></p>
                    <p>
                        <a href={"/#/forgotPassword"}>Glemt passord</a>
                    </p>
                </div>
              </div>
            </Col>

            <Col sm={2} md={3} />
          </Form>
        </Grid>
      </div>
    );
  } //end method

 save = async () => {
     if (confirm('Ved å logge inn på Hverdagshelt.no godtar du at vi lagrer ' +
         'cookies med informasjon om brukernavnet  ditt og hvilken type bruker du er (privat / bedirft')) {
         await userService
             .getUserLogin(this.state.email)
             .then(response => {
                 if (response[0].active === 1) {
                     this.setState({
                         countyId: response[0].countyId,
                         storedPassword: response[0].password
                     });
                 } else {
                     this.setState({blocked: true});
                     return;
                 }

                 bcrypt.compare(this.state.password, response[0].password, (err, res) => {
                     if (res) {
                         userService.login({userMail: response[0].mail, typeId: response[0].typeName}).then(r => {
                             let token = r.jwt;
                             window.localStorage.setItem('userToken', token);
                             console.log('login in success');
                             userService.getCurrentUser().then(r3 => {
                                 window.sessionStorage.setItem('countyId', r3[0].countyId);
                                 window.sessionStorage.setItem('countyName', r3[0].county);
                             });

                             window.location.reload();
                             history.push('/wizardForm');
                         }).catch((error: Error) => confirm(error.message));
                     } else { //check if the email is a company email
                         userService.getCompanyLogin(this.state.email).then(r => {
                             bcrypt.compare(this.state.password, r[0].password, (err, res) => {
                                 if (res) {
                                     userService.login({userMail: r[0].mail, typeId: 'Company'}).then(r => {
                                         let token = r.jwt;
                                         window.localStorage.setItem('userToken', token);
                                         console.log('login in success');
                                         console.log('hei');

                                         window.location.reload();
                                         history.push('/wizardForm');
                                     }).catch((error: Error) => confirm(error.message));
                                 } else {
                                     this.setState({
                                         error: true
                                     });
                                 }//end condition
                             });
                         }).catch((error: Error) => {
                             console.log(error);
                             this.setState({
                                 error: true
                             });
                         });
                     }//end condition
                 });
             }).catch((error: Error) => {
                 userService.getCompanyLogin(this.state.email).then(r => {
                     bcrypt.compare(this.state.password, r[0].password, (err, res) => {
                         if (res) {
                             userService.login({userMail: this.state.email, typeId: 'Company'}).then(r => {
                                 let token = r.jwt;
                                 window.localStorage.setItem('userToken', token);
                                 console.log('login in success');


                                 window.location.reload();
                                 history.push('/wizardForm');
                             }).catch((error: Error) => {
                                 console.log(error);
                                 this.setState({
                                     error: true
                                 });
                             });
                         } else {
                             this.setState({
                                 error: true
                             });
                         }//end condition
                     });
                 }).catch((error: Error) => {
                     console.log(error);
                     this.setState({
                         error: true
                     });
                 });
             });
     } else {
        window.location.reload()
     }
 }//end method
} //end class