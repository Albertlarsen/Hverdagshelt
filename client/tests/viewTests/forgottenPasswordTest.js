// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {ForgottenPassword} from "../../src/views/login/forgottenPassword";

describe('Test for ForgottenPassword view', () => {
    const wrapper = shallow(<ForgottenPassword />);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});