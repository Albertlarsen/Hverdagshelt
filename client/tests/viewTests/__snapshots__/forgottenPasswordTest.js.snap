// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ForgottenPassword view renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgottenPassword />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Glemt passord
        </h2>,
        <p>
          Skriv inn eposten din og få tilsendt et nytt passord på mail.
        </p>,
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
        >
          <FormGroup
            bsClass="form-group"
          >
            <div
              className="textfield"
            >
              <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                placeholder="Epost"
                value=""
              />
              <FormControlFeedback
                bsClass="form-control-feedback"
                bsRole="feedback"
              />
            </div>
          </FormGroup>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="primary"
            disabled={false}
            onClick={[Function]}
          >
            Tilbakestill passord
          </Button>
        </Form>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Glemt passord",
        },
        "ref": null,
        "rendered": "Glemt passord",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Skriv inn eposten din og få tilsendt et nytt passord på mail.",
        },
        "ref": null,
        "rendered": "Skriv inn eposten din og få tilsendt et nytt passord på mail.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": Array [
            <FormGroup
              bsClass="form-group"
            >
              <div
                className="textfield"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Epost"
                  value=""
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </div>
            </FormGroup>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              onClick={[Function]}
            >
              Tilbakestill passord
            </Button>,
          ],
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": <div
                className="textfield"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Epost"
                  value=""
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </div>,
              "validationState": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Epost"
                    value=""
                  />,
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />,
                ],
                "className": "textfield",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "componentClass": "input",
                    "onChange": [Function],
                    "placeholder": "Epost",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control-feedback",
                    "bsRole": "feedback",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "primary",
              "children": "Tilbakestill passord",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Tilbakestill passord",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Glemt passord
          </h2>,
          <p>
            Skriv inn eposten din og få tilsendt et nytt passord på mail.
          </p>,
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
          >
            <FormGroup
              bsClass="form-group"
            >
              <div
                className="textfield"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Epost"
                  value=""
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </div>
            </FormGroup>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              onClick={[Function]}
            >
              Tilbakestill passord
            </Button>
          </Form>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Glemt passord",
          },
          "ref": null,
          "rendered": "Glemt passord",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Skriv inn eposten din og få tilsendt et nytt passord på mail.",
          },
          "ref": null,
          "rendered": "Skriv inn eposten din og få tilsendt et nytt passord på mail.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form",
            "children": Array [
              <FormGroup
                bsClass="form-group"
              >
                <div
                  className="textfield"
                >
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Epost"
                    value=""
                  />
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </div>
              </FormGroup>,
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                onClick={[Function]}
              >
                Tilbakestill passord
              </Button>,
            ],
            "componentClass": "form",
            "horizontal": false,
            "inline": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": <div
                  className="textfield"
                >
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Epost"
                    value=""
                  />
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </div>,
                "validationState": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      onChange={[Function]}
                      placeholder="Epost"
                      value=""
                    />,
                    <FormControlFeedback
                      bsClass="form-control-feedback"
                      bsRole="feedback"
                    />,
                  ],
                  "className": "textfield",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-control",
                      "componentClass": "input",
                      "onChange": [Function],
                      "placeholder": "Epost",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-control-feedback",
                      "bsRole": "feedback",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "primary",
                "children": "Tilbakestill passord",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Tilbakestill passord",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
